spec/integ/matrix-client-event-timeline.spec.ts

```diff
@@ -925,6 +974,231 @@ describe("MatrixClient event timelines", function() {
         });
     });
 
+    describe("paginateEventTimeline for thread list timeline", function() {
+        async function flushHttp<T>(promise: Promise<T>): Promise<T> {
+            return Promise.all([promise, httpBackend.flushAllExpected()]).then(([result]) => result);
+        }
+
+        describe("with server compatibility", function() {
+            async function testPagination(timelineSet: EventTimelineSet, direction: Direction) {
+                const RANDOM_TOKEN = "7280349c7bee430f91defe2a38a0a08c";
+                function respondToThreads() {
+                    httpBackend.when("GET", encodeUri("/_matrix/client/r0/rooms/$roomId/threads", {
+                        $roomId: roomId,
+                    })).respond(200, {
+                        chunk: [STABLE_THREAD_ROOT],
+                        state: [],
+                        next_batch: RANDOM_TOKEN,
+                    });
+                }
+                function respondToContext() {
+                    httpBackend.when("GET", encodeUri("/_matrix/client/r0/rooms/$roomId/context/$eventId", {
+                        $roomId: roomId,
+                        $eventId: STABLE_THREAD_ROOT.event_id!,
+                    })).respond(200, {
+                        end: "",
+                        start: "",
+                        state: [],
+                        events_before: [],
+                        events_after: [],
+                        event: STABLE_THREAD_ROOT,
+                    });
+                }
+
+                respondToContext();
+                await flushHttp(client.getEventTimeline(timelineSet, STABLE_THREAD_ROOT.event_id!));
+                respondToThreads();
+                const timeline = await flushHttp(client.getLatestTimeline(timelineSet));
+                expect(timeline).not.toBeNull();
+
+                respondToThreads();
+                const success = await flushHttp(client.paginateEventTimeline(timeline!, {
+                    backwards: direction === Direction.Backward,
+                }));
+                expect(success).toBeTruthy();
+                expect(timeline!.getEvents().length).toEqual(1);
+                expect(timeline!.getEvents()[0].event).toEqual(STABLE_THREAD_ROOT);
+                expect(timeline!.getPaginationToken(direction)).toEqual(RANDOM_TOKEN);
+            }
+
+            it("should allow you to paginate all threads backwards", async function() {
+                // @ts-ignore
+                client.clientOpts.experimentalThreadSupport = true;
+                Thread.setServerSideSupport(FeatureSupport.Experimental);
+                Thread.setServerSideListSupport(FeatureSupport.Stable);
```

Comment:
This could be put in a `beforeEach` since it's repeated in all test in this block